### PR TITLE
Add: Cloud icon for cloud scores in the file menu for non-macOS platf…

### PR DESCRIPTION
**Pull request by nasehim7**
Resolves: #18014 (For non-Mac OS)

For more information, refer this discussion thread (archived PR): https://github.com/musescore/MuseScore/pull/21802

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [x] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

File: src/appshell/qml/platform/PlatformMenuBar.qml [hunk 1]
```diff
@@ -100,6 +100,7 @@ Item {
         menuItem.checkable = itemInfo.checkable
         menuItem.separator = !Boolean(itemInfo.title)
         menuItem.role = itemInfo.role
+        menuItem.icon.mask = true
```
Comment (cbjeukendrup) [inline src/appshell/qml/platform/PlatformMenuBar.qml:103]: I think this change shouldn't be included in this PR

Comment (nasehim7) [reply src/appshell/qml/platform/PlatformMenuBar.qml:103]: Well now that I think about it again -> here we are actually updating the UiAction properties but in the QML we are updating the menuItem property so it's not needed. Correct?

**Review by cbjeukendrup — APPROVED** [review]

**Comment by cbjeukendrup** [comment]
@jessjwilliamson This PR should solve the linked issue, but for now only on Windows and Linux, so it would be great if you could test it on one of these OSs. For macOS, it turns out to be (much) more difficult to fix, and it would be a lower priority because we generally don't use icons at all in menus on macOS, so we postponed that; perhaps it will still happen someday in the future.

**Comment by zacjansheski** [comment]
Tested on MacOS 14, Windows 11, Ubuntu 22.04.3. Approved
